hibernate-core/src/main/java/org/hibernate/cfg/AnnotatedColumn.java

```diff
@@ -531,7 +539,7 @@ public void forceNotNull() {
 		mappingColumn.setNullable( false );
 	}
 
-	public static AnnotatedColumn[] buildFormulaFromAnnotation(
+	public static AnnotatedColumns buildFormulaFromAnnotation(
```

Comment:
And retroactively notating stuff as internal is not long term supportable.

Reply:
Here, e.g., just make new methods returning `AnnotatedColumns` and have these return `AnnotatedColumns#getColumns`

Reply:
AGAIK, these classes are only used internally by `AnnotationBinder`, and they don't survive beyond its lifecycle. I can't see a way that a user could even get a reference to one of these things, let alone integrate with or extend any of this stuff. `AnnotationBinder` itself just doesn't expose any of this stuff.

Unless I'm missing something.

Reply:
Not sure what "references" you mean.  I commented on some static methods that are used in building parts of the boot model.  Other things (tooling, e.g.) can potentially use these.  We just don't know

Reply:
Please just make it backwards compatible.  Its really very easy here

Reply:
P.S. An important thing to point out here is that the types `AnnotatedColumn` and `AnnotatedJoinColumn` did not exist in Hibernate 5. This PR changes the API of what are in principle brand-new internal implementation classes that were "introduced" in 6.0. 

And since, as far as I understand, neither 6.0 nor 6.1 have yet been released in any product, there's actually _no breakage to supported software in this PR_.

I therefore think it's very important that we get https://hibernate.atlassian.net/browse/HHH-15650 sorted out before Hibernate 6 goes into EAP.

Reply:
Steve, Sanne, and I discussed this offline from folks in productization and we have the go-ahead to do what is proposed in HHH-15650. So I'm going to merge this.